packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -146,6 +168,24 @@ function getNodeSchemaType(nodeSchema: TreeNodeStoredSchema): SchemaFactoryNodeK
 	throwUnsupportedNodeType(nodeSchema.constructor.name);
 }
 
+function getViewNodeSchemaType(schema: TreeNodeSchema): SchemaFactoryNodeKind {
+	switch (schema.kind) {
+		case NodeKind.Leaf: {
+			return "leaf";
+		}
+		case NodeKind.Map: {
```

Comment:
Josh recently added the record type, which works like Maps and should be handled here. Ensure it is handled here and has test coverage.

Reply:
This seems like a bug and I don't think it should be addressed in this pr since this is meant to purely be a refactor. I made a bug here: https://dev.azure.com/fluidframework/internal/_workitems/edit/43720 and assigned it to Josh for now but I can also deal with it later if Josh is busy. @Josmithr FYI